drivers/adc/adc_stm32.c

```diff
@@ -557,7 +557,7 @@ static int adc_stm32_init(struct device *dev)
 		LOG_ERR("ADC clock rate get error.");
 	}
 
-	wait_cycles = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / adc_rate *
```

Comment:
Use `SystemCoreClock` instead of `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC`.
`SystemCoreClock` is updated after clock configuration, which won't happen on `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` or` DT_CPUS_CPU_0_CLOCK_FREQUENCY`

Reply:
Fixed.